types/aframe/index.d.ts

```diff
@@ -0,0 +1,324 @@
+// Type definitions for AFRAME 0.5
+// Project: https://aframe.io/
+// Definitions by: Paul Shannon <https://github.com/devpaul>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+/// <reference types="three" />
+/// <reference types="tweenjs" />
+
+// Globals
+declare var AFRAME: AFrame.AFrameGlobal;
+declare var hasNativeWebVRImplementation: boolean;
+
+interface NodeSelector {
+	querySelector(selectors: 'a-scene'): AFrame.Scene;
+	querySelector<T extends AFrame.Entity<any>>(selectors: string): T;
+	querySelectorAll(selectors: string): NodeListOf<AFrame.Entity<any> | Element>;
+}
+
+interface Document {
+	createElement(tagName: string): AFrame.Entity;
+}
+
+// Interfaces
+declare namespace AFrame {
+	interface ObjectMap<T = any> {
+		[ key: string ]: T;
+	}
+
+	interface AFrameGlobal {
+		AEntity: Entity;
+		ANode: ANode;
+		AScene: Scene;
+		components: { [ key: string ]: ComponentDescriptor };
+		geometries: { [ key: string ]: GeometryDescriptor };
+		primitives: { [ key: string ]: Entity };
+		registerComponent(name: string, component: ComponentDefinition): ComponentConstructor;
+		registerElement(name: string, element: ANode): void;
+		registerGeometry(name: string, geometery: THREE.Geometry): Geometry;
+		registerPrimitive(name: string, primitive: PrimitiveDefinition): void;
+		registerShader(name: string, shader: any): void;
+		registerSystem(name: string, definition: SystemDefinition): void;
+		schema: SchemaUtils;
+		shaders: { [ key: string ]: ShaderDescriptor };
+		systems: { [key: string]: System };
+		THREE: any; // TODO THREE
```

Comment:
It took some time to figure out, but if I use `typeof THREE` it seems to work. I'll update.